sw/device/tests/otbn/otbn_rsa_test.c

```diff
@@ -0,0 +1,903 @@
+  ctx->otbnptr_modulus = _otbn_app_rsa_modulus - _otbn_app_rsa__dmem_start;
+  ctx->otbnptr_exp = _otbn_app_rsa_exp - _otbn_app_rsa__dmem_start;
+
+  if (dif_otbn_init(dif_config, &ctx->dif) != kDifOtbnOk) {
+    return kOtbnRsaError;
+  }
+
+  return kOtbnRsaOk;
+}
+
+/**
+ * (Re-)load the RSA application into OTBN.
+ *
+ * Load the application image with both instruction and data segments into OTBN.
+ *
+ * @param ctx the context object
+ * @return kOtbnRsaOk if the operation was successful, kOtbnRsaError otherwise
+ */
+static otbn_rsa_result_t otbn_rsa_load_app(otbn_rsa_t *ctx) {
+  if (ctx->app->imem_end <= ctx->app->imem_start) {
+    return kOtbnRsaBadArg;
+  }
+  if (ctx->app->dmem_end < ctx->app->dmem_start) {
+    return kOtbnRsaBadArg;
+  }
+
+  const size_t imem_size = ctx->app->imem_end - ctx->app->imem_start;
+  const size_t dmem_size = ctx->app->dmem_end - ctx->app->dmem_start;
+
+  // Instruction and data memory images must be multiples of 32b words.
+  if (imem_size % 4 != 0 || dmem_size % 4 != 0) {
+    return kOtbnRsaBadArg;
+  }
+
+  if (dif_otbn_imem_write(&ctx->dif, 0, ctx->app->imem_start, imem_size) !=
+      kDifOtbnOk) {
+    return kOtbnRsaError;
+  }
+
+  if (dmem_size > 0) {
+    if (dif_otbn_dmem_write(&ctx->dif, 0, ctx->app->dmem_start, dmem_size) !=
+        kDifOtbnOk) {
+      return kOtbnRsaError;
+    }
+  }
+
+  ctx->is_loaded = true;
+  return kOtbnRsaOk;
+}
+
+/**
+ * Encrypt a message with RSA
+ *
+ * @param ctx the context object
+ * @param modulus the modulus n
+ * @param in the plaintext message
+ * @param out the encrypted message
+ * @param size_bytes the size of all buffers in bytes, i.e. the key/modulus
+ *                   length (i.e. 128 for RSA 1024). Valid range: 32..512 in
+ *                   32 byte-steps (i.e. RSA 256 to RSA 4096).
+ * @return kOtbnRsaOk if the operation was successful, kOtbnRsaError otherwise
+ */
+static otbn_rsa_result_t otbn_rsa_encrypt(otbn_rsa_t *ctx,
+                                          const uint8_t *modulus,
+                                          const uint8_t *in, uint8_t *out,
+                                          size_t size_bytes) {
+  if (ctx == NULL || !ctx->is_loaded) {
+    return kOtbnRsaBadArg;
+  }
+
+  if (size_bytes % 32 != 0) {
+    return kOtbnRsaBadArg;
+  }
+
+  uint32_t n_limbs = size_bytes / 32;
+  if (n_limbs == 0 || n_limbs > 16) {
+    return kOtbnRsaBadArg;
+  }
+
+  // Write input arguments
+  if (dif_otbn_dmem_write(&ctx->dif, ctx->otbnptr_n_limbs, &n_limbs,
+                          sizeof(n_limbs)) != kDifOtbnOk) {
+    return kOtbnRsaError;
+  }
+
+  if (dif_otbn_dmem_write(&ctx->dif, ctx->otbnptr_modulus, modulus,
+                          size_bytes) != kDifOtbnOk) {
+    return kOtbnRsaError;
+  }
+
+  if (dif_otbn_dmem_write(&ctx->dif, ctx->otbnptr_in, in, size_bytes) !=
+      kDifOtbnOk) {
+    return kOtbnRsaError;
+  }
+
+  // Call OTBN to perform operation
+  otbn_rsa_result_t rv;
```

Comment:
Don't pre-declare variables, and avoid initialisms like rv; prefer `err` or `result` or similar.